connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetricsRegistry.java

```diff
@@ -0,0 +1,283 @@
+    public static final String SOURCE_TASK_GROUP_NAME = "source-task-metrics";
+    public static final String SINK_TASK_GROUP_NAME = "sink-task-metrics";
+
+    private final List<MetricNameTemplate> allTemplates = new ArrayList<>();
+    public final MetricNameTemplate connectorStatusRunning;
+    public final MetricNameTemplate connectorStatusPaused;
+    public final MetricNameTemplate connectorStatusFailed;
+    public final MetricNameTemplate taskStatusUnassigned;
+    public final MetricNameTemplate taskStatusRunning;
+    public final MetricNameTemplate taskStatusPaused;
+    public final MetricNameTemplate taskStatusFailed;
+    public final MetricNameTemplate taskStatusDestroyed;
+    public final MetricNameTemplate taskRunningRatio;
+    public final MetricNameTemplate taskPauseRatio;
+    public final MetricNameTemplate taskCommitTimeMax;
+    public final MetricNameTemplate taskCommitTimeAvg;
+    public final MetricNameTemplate taskBatchSizeMax;
+    public final MetricNameTemplate taskBatchSizeAvg;
+    public final MetricNameTemplate taskCommitFailurePercentage;
+    public final MetricNameTemplate taskCommitSuccessPercentage;
+    public final MetricNameTemplate sourceRecordPollRate;
+    public final MetricNameTemplate sourceRecordPollTotal;
+    public final MetricNameTemplate sourceRecordWriteRate;
+    public final MetricNameTemplate sourceRecordWriteTotal;
+    public final MetricNameTemplate sourceRecordPollBatchTimeMax;
+    public final MetricNameTemplate sourceRecordPollBatchTimeAvg;
+    public final MetricNameTemplate sinkRecordReadRate;
+    public final MetricNameTemplate sinkRecordReadTotal;
+    public final MetricNameTemplate sinkRecordSendRate;
+    public final MetricNameTemplate sinkRecordSendTotal;
+    public final MetricNameTemplate sinkRecordLagMax;
+    public final MetricNameTemplate sinkRecordPartitionCount;
+    public final MetricNameTemplate sinkRecordOffsetCommitSeqNum;
+    public final MetricNameTemplate sinkRecordOffsetCommitCompletionRate;
+    public final MetricNameTemplate sinkRecordOffsetCommitCompletionTotal;
+    public final MetricNameTemplate sinkRecordOffsetCommitSkipRate;
+    public final MetricNameTemplate sinkRecordOffsetCommitSkipTotal;
+    public final MetricNameTemplate sinkRecordPutBatchTimeMax;
+    public final MetricNameTemplate sinkRecordPutBatchTimeAvg;
+    public final MetricNameTemplate sinkRecordTopicPartitionLag;
+    public final MetricNameTemplate sinkRecordTopicPartitionLagMax;
+    public final MetricNameTemplate sinkRecordTopicPartitionLagAvg;
+
+    public ConnectMetricsRegistry() {
+        this(new LinkedHashSet<String>());
+    }
+
+    public ConnectMetricsRegistry(Set<String> tags) {
+        /***** Connector level *****/
+        Set<String> connectorTags = new LinkedHashSet<>(tags);
+        connectorTags.add(CONNECTOR_TAG_NAME);
+
+        connectorStatusRunning = createTemplate("status-running", CONNECTOR_GROUP_NAME,
```

Comment:
fyi, the pr for non-double stats got merged, so we could switch this back to the original design / fill in the missing string metric

Reply:
Is there any point in keeping the separate indicator metrics? How about I do these with the next PR for KAFKA-5903?